php-checks/src/main/java/org/sonar/php/checks/PHP5DeprecatedFunctionUsageCheck.java

```diff
@@ -63,56 +68,53 @@
-    AstNode nextNode = memberExpr.getNextAstNode();
-    String calledFunctionName = memberExpr.getTokenOriginalValue();
-    String replacement = NEW_BY_DEPRECATED_FUNCTIONS.get(calledFunctionName);
-
-
-    if (nextNode != null && nextNode.is(PHPGrammar.FUNCTION_CALL_PARAMETER_LIST)) {
+  protected void createIssue(FunctionCallTree tree) {
+    String functionName = functionName(tree.callee());
 
-      if ("setlocale".equals(calledFunctionName)) {
-        String category = getCategoryFromStringLiteralParameter(nextNode);
+    if (SET_LOCALE_FUNCTION.equals(functionName)) {
+      checkLocalCategoryArgument(tree, tree.arguments());
 
-        if (category != null) {
-          getContext().createLineViolation(this, "Use the \"{0}\" constant instead of a string literal.", astNode, category);
-        }
-
-      } else if (replacement != null) {
-        getContext().createLineViolation(this, buildMessage(calledFunctionName, replacement), astNode);
-      }
+    } else {
+      context().newIssue(this.KEY, buildMessage(functionName)).tree(tree.callee());
     }
-  }
 
-  private String getCategoryFromStringLiteralParameter(AstNode parameterList) {
-    String firstParam = parameterList.getFirstChild(PHPPunctuator.LPARENTHESIS).getNextAstNode().getTokenOriginalValue();
+  }
 
-    if (firstParam.startsWith("\"") && firstParam.endsWith("\"") || firstParam.startsWith("'") && firstParam.endsWith("'")) {
-      String category = firstParam.substring(1, firstParam.length() - 1);
+  /**
+   * Build issue message depending on the presence of a replacement function of not.
+   */
+  private static String buildMessage(String functionName) {
+    String replacement = NEW_BY_DEPRECATED_FUNCTIONS.get(functionName);
 
-      if (LOCALE_CATEGORY_CONSTANTS.contains(category)) {
-        return category;
-      }
-    }
-    return null;
+    return replacement.isEmpty() ?
+      String.format("Remove this \"%s()\" call.", functionName) :
+      String.format("Replace this \"%s()\" call with a call to \"%s\".", functionName, replacement);
   }
 
-  private String buildMessage(String calledFunctionName, String replacement) {
-    if (replacement.isEmpty()) {
-      return "Remove this \"" + calledFunctionName + "()\" call.";
+  /**
+   * Raise an issue if the local category is passed as a String.
+   */
+  private void checkLocalCategoryArgument(FunctionCallTree tree, SeparatedList<ExpressionTree> arguments) {
+    if (!arguments.isEmpty() && arguments.get(0).is(Kind.REGULAR_STRING_LITERAL)) {
+      String firstArg = ((LiteralTree) arguments.get(0)).value();
+      String localCategory = firstArg.substring(1, firstArg.length() - 1);
 
-    } else {
-      return "Replace this \"" + calledFunctionName + "()\" call with a call to \"" + replacement + "\".";
+      if (LOCALE_CATEGORY_CONSTANTS.contains(localCategory)) {
+        context().newIssue(this.KEY, String.format(SET_LOCAL_ARG_MESSAGE, localCategory)).tree(tree.callee());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2209)